metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/BasicStellarTest.java

```diff
@@ -251,6 +251,142 @@ public void testNull() {
+    }
+
+    //  any > >= < <= is false
+    {
+      String query = "NaN > 5";
+      Assert.assertFalse(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN < 5";
+      Assert.assertFalse(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN >= 5";
+      Assert.assertFalse(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN <= 5";
+      Assert.assertFalse(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN > NaN";
+      Assert.assertFalse(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN < NaN";
+      Assert.assertFalse(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN >= NaN";
+      Assert.assertFalse(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN <= NaN";
+      Assert.assertFalse(runPredicate(query,new HashMap<>()));
+    }
+
+    // all operations
+    {
+      String query = "(5 + NaN) != NaN";
+      Assert.assertTrue(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "5 + NaN";
+      Assert.assertTrue(run(query,new HashMap<>()).toString().equals("NaN"));
+    }
+    {
+      String query = "(5 - NaN) != NaN";
+      Assert.assertTrue(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "5 - NaN";
+      Assert.assertTrue(run(query,new HashMap<>()).toString().equals("NaN"));
+    }
+    {
+      String query = "(5 / NaN) != NaN";
+      Assert.assertTrue(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "5 / NaN";
+      Assert.assertTrue(run(query,new HashMap<>()).toString().equals("NaN"));
+    }
+    {
+      String query = "(5 * NaN) != NaN";
+      Assert.assertTrue(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "5 * NaN";
+      Assert.assertTrue(run(query,new HashMap<>()).toString().equals("NaN"));
+    }
+    {
+      String query = "(NaN + NaN) != NaN";
+      Assert.assertTrue(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN + NaN";
+      Assert.assertTrue(run(query,new HashMap<>()).toString().equals("NaN"));
+    }
+    {
+      String query = "(NaN - NaN) != NaN";
+      Assert.assertTrue(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN - NaN";
+      Assert.assertTrue(run(query,new HashMap<>()).toString().equals("NaN"));
+    }
+    {
+      String query = "(NaN * NaN) != NaN";
+      Assert.assertTrue(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN * NaN";
+      Assert.assertTrue(run(query,new HashMap<>()).toString().equals("NaN"));
+    }
+    {
+      String query = "(NaN / NaN) != NaN";
+      Assert.assertTrue(runPredicate(query,new HashMap<>()));
+    }
+    {
+      String query = "NaN / NaN";
+      Assert.assertTrue(run(query,new HashMap<>()).toString().equals("NaN"));
+    }
```

Comment:
Can you add a test case for 0.0/0.0 is Nan?  It's properly calculated based on my playing around, but it's an important enough case that I think it should show up here.

Reply:
done